phoenix-pherf/src/main/java/org/apache/phoenix/pherf/configuration/XMLConfigParser.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ *   or more contributor license agreements.  See the NOTICE file
+ *   distributed with this work for additional information
+ *   regarding copyright ownership.  The ASF licenses this file
+ *   to you under the Apache License, Version 2.0 (the
+ *   "License"); you may not use this file except in compliance
+ *   with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.apache.phoenix.pherf.configuration;
+
+import org.apache.phoenix.pherf.PherfConstants;
+import org.apache.phoenix.pherf.exception.FileLoaderException;
+import org.apache.phoenix.pherf.util.ResourceList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+
+public class XMLConfigParser {
+
+    private static final Logger logger = LoggerFactory.getLogger(XMLConfigParser.class);
+    private String filePattern;
+    private List<DataModel> dataModels;
+    private List<Scenario> scenarios = null;
+    private ResourceList resourceList;
+    private Collection<Path> paths = null;            // Only populate once. It may be slow of CP is large.
+
+    public XMLConfigParser(String pattern) throws Exception{
+        init(pattern);
+    }
+
+    public List<DataModel> getDataModels() {
+        return dataModels;
+    }
+
+    public Collection<Path> getPaths(String strPattern) throws Exception{
```

Comment:
I'll synchronize it. Threads may actually call it. Though I have not seen any concurrency bugs due to it.